tests/test_noise.py

```diff
@@ -130,3 +143,122 @@ def pinkish_spectrum(x, y):
     # Check we are achieving an estimate near the target
     logger.debug(f"Estimated Noise Variance {estimated_noise_var}")
     assert np.isclose(estimated_noise_var, target_noise_variance, rtol=0.1)
+
+    # Check sampling yields an estimate near target.
+    sample_n = 16
+    sample_res = 32
+    im_zeros = Image(np.zeros((sample_n, sample_res, sample_res)))
+    im_noise_sample = sim_fixture.noise_adder._forward(im_zeros, range(sample_n))
+    sampled_noise_var = np.var(im_noise_sample.asnumpy())
+
+    logger.debug(f"Sampled Noise Variance {sampled_noise_var}")
+    assert np.isclose(sampled_noise_var, target_noise_variance, rtol=0.1)
+
+
+@pytest.mark.parametrize(
+    "target_noise_variance", VARS, ids=lambda param: f"var={param}"
+)
+def test_from_snr_white(sim_fixture, target_noise_variance):
+    """
+    Test that prescribing noise directly by var and  by `from_snr`,
+    are close for a variety of paramaters.
+    """
+
+    # First add an explicit amount of noise to the base simulation,
+    sim_fixture.noise_adder = WhiteNoiseAdder(var=target_noise_variance)
+    # and compute the resulting snr of the sim.
+    target_snr = sim_fixture.estimate_snr()
+
+    # Attempt to create a new simulation at this `target_snr`
+    # For unit testing, we will use `sim_fixture`'s volume,
+    #   but the new Simulation instance should yield different projections.
+    sim_from_snr = Simulation(
+        vols=sim_fixture.vols,  # Force the previously generated volume.
+        n=sim_fixture.n,
+        offsets=0,
+        amplitudes=1.0,
+        noise_adder=WhiteNoiseAdder.from_snr(target_snr),
+    )
+
+    # Check we're within 5% of explicit target
+    logger.info(
+        "sim_from_snr.noise_adder.noise_var, target_noise_variance ="
+        f" {sim_from_snr.noise_adder.noise_var}, {target_noise_variance}"
+    )
+    assert np.isclose(
+        sim_from_snr.noise_adder.noise_var, target_noise_variance, rtol=0.05
+    )
+
+    # Compare with WhiteNoiseEstimator consuming sim_from_snr
+    noise_estimator = WhiteNoiseEstimator(sim_from_snr, batchSize=512)
+    est_noise_variance = noise_estimator.estimate()
+    logger.info(
+        "est_noise_variance, target_noise_variance ="
+        f" {est_noise_variance}, {target_noise_variance}"
+    )
+
+    # Check we're within 5%
+    assert np.isclose(est_noise_variance, target_noise_variance, rtol=0.05)
+
+
+@pytest.mark.parametrize(
+    "target_noise_variance", VARS, ids=lambda param: f"var={param}"
+)
+def test_from_snr_pink_iso(sim_fixture, target_noise_variance):
+    """
+    Test that prescribing noise directly by var and  by `from_snr`,
+    are close for a variety of paramaters.
+    """
```

Comment:
I'm confused about how this test and the one below are prescribing noise `from_snr`. What am I missing?

Reply:
No worries. I just forgot the change the name.

They are not currently, but I think they might have been before...

Currently they use `target_noise_variance` parameter.  This test eventually belongs in another section.  Its possible in the future we could do from_snr for other adders as well.